mkosi/__init__.py

```diff
@@ -148,13 +148,36 @@ class MkosiException(Exception):
     """Leads to sys.exit"""
 
 
-def print_step(text: str) -> None:
-    sys.stderr.write("‣ \033[0;1;39m" + text + "\033[0m\n")
+class MkosiPrinter():
+    out_file = sys.stderr
+    isatty = out_file.isatty()
+
+    bold = "\033[0;1;39m" if isatty else ""
+    red = "\033[31;1m" if isatty else ""
+    reset = "\033[0m" if isatty else ""
+
+    prefix = "‣ "
+
+    @classmethod
+    def _p(cls, text: str) -> None:
```

Comment:
Saving letters doesn't matter here, better to call it _print

Reply:
done